### PR TITLE
Testing/main page

### LINKED ISSUE #2: Random Joke Button
### User Story
- [x] As a joke lover, I want to click a button to generate a new random joke, providing an endless supply of humor.
- [x] If the server is unable to fetch a new joke, I should see an error message.

### Acceptance Criteria
- [x] Page should contain a "New Joke" button.
- [x] When clicked, button should fetch and display new joke.
- [x] The old joke should disappear.

### LINKED ISSUE #4: Main Page Cypress
Cypress tests
- [x] Navigation to main page
- [x] Intercept fetch request and simulate 200, 400, 500 responses
- [x] Check that joke element has rendered and contains correct text

### PR DESCRIPTION
**Pull request by kirkhauck**
# Description

This branch covers the happy and sad path testing for the home page. It intercepts the fetch requests to the dad joke API and stubs a couple mock jokes for the tests. It tests to make sure the page elements render, that a new joke generates when the button is clicked, and that an error message appears for bad fetch requests.

## Type of change

- [x] Cypress tests

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ]  local host
- [ ]  dev tools
- [ ]  terminal
- [x] Cypress

# Related Issues:

- [x]  closes #2 closes #4 

# Checklist:

- [x]  My code follows the Turing's guidelines of this project
- [x]  I have performed a self-review of my own code
- [x]  No console error messages

### CODE REVIEW

File: cypress/e2e/home-page-spec.cy.ts
Comment (Kanderson58) [inline cypress/e2e/home-page-spec.cy.ts:1]: Great work handling the typescript dependencies!  Great happy/sad path testing.

**Review by Kanderson58 — COMMENTED** [review]
Goal for later on:  Move sample data into fixtures

File: cypress/e2e/home-page-spec.cy.ts
Comment (LSeward0421) [inline cypress/e2e/home-page-spec.cy.ts:1]: Awesome work! Thanks for covering the sad paths too!!!

**Review by LSeward0421 — APPROVED** [review]
Awesome work!

**Pull request merged**